docs/common-pitfalls.md

```diff
@@ -0,0 +1,42 @@
+# Common Pitfalls
+
+## AVA in Docker
+
+If you run AVA in Docker as part of your CI, you need to fix the appropriate environment variables. Specifically, adding `-e CI=true` in the `docker exec` command. See [https://github.com/avajs/ava/issues/751](#751).
+
+AVA uses [is-ci](https://github.com/watson/is-ci) to decide if it's in a CI environment or not using [these variables](https://github.com/watson/is-ci/blob/master/index.js).
+
+## AVA and connected client limits
+
+You may be using a service that only allows a limited number of concurrent connections. For example, many database-as-a-service businesses offer a free plan with a limit on how many clients can be using it at the same time. AVA can hit those limits as it runs multiple processes, but well-written services should emit an error or throttle in those cases. If the one you're using doesn't, the tests will hang.
+
+Use the `concurrency` flag to limit the number of processes ran. For example, if your service plan allows 5 clients, you should run AVA with `concurrency=5` or less.
+
+## Async operations
```

Comment:
I opened an issue about this: https://github.com/avajs/eslint-plugin-ava/issues/122

We should really try to detect this at runtime and in our ESLint plugin too.